test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use log::error;
+use serde::{Deserialize, Deserializer};
+use serde_json::Value;
+use up_rust::{
+    Data, UAttributes, UAuthority, UCode, UEntity, UMessage, UMessageType, UPayload,
+    UPayloadFormat, UPriority, UResource, UUri, UUID,
+};
+
+use protobuf::{Enum, MessageField, SpecialFields};
+
+pub fn convert_json_to_jsonstring<T: serde::Serialize>(value: &T) -> String {
+    if let Ok(json_string) = serde_json::to_string(value) {
+        json_string
+    } else {
+        // Handle the error
+        error!("Error: Failed to convert to JSON string");
+        "None".to_owned()
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct WrapperUUri(pub UUri);
+impl<'de> Deserialize<'de> for WrapperUUri {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+        let mut authority = UAuthority::new();
+        let mut uuri: UUri = UUri::new();
+        let mut entity = UEntity::new();
+        let mut resource = UResource::new();
+        //update authority
+
+        if let Some(authority_value) = value
```

Comment:
Suggest reading my suggestions I made over in these [comments](https://github.com/eclipse-uprotocol/up-tck/pull/23#discussion_r1566164311) to make this more ergonomic and avoid having mutable variables we carry throughout the function:
```rust
let mut authority = UAuthority::new();
let mut uuri: UUri = UUri::new();
let mut entity = UEntity::new();
let mut resource = UResource::new();
```